src/main/scala/org/apache/mesos/chronos/scheduler/api/DependentJobResource.scala

```diff
@@ -57,10 +57,16 @@ class DependentJobResource @Inject()(
         val parents = jobGraph.parentJobs(newJob)
         oldJob match {
           case j: DependencyBasedJob =>
-            val oldParents = jobGraph.parentJobs(j)
-            oldParents.map(x => jobGraph.removeDependency(x.name, oldJob.name))
+            val newParentNames = parents.map(_.name)
+            val oldParentNames = jobGraph.parentJobs(j).map(_.name)
+
+            if (newParentNames != oldParentNames) {
+              oldParentNames.foreach(jobGraph.removeDependency(_, oldJob.name))
+              newParentNames.foreach(jobGraph.addDependency(_, newJob.name))
+            }
             jobScheduler.removeSchedule(j)
           case j: ScheduleBasedJob =>
```

Comment:
This makes sense, I think this line: ``parents.foreach(p => jobGraph.addDependency(p.name, newJob.name))`` needs to be outside the case statement as it was before, though. This is because both types of jobs can have child jobs. DependencyBasedJobs will be the only one with parents.